File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> uset1;
        unordered_set<int> uset2;

        for (int n : nums1) {
            uset1.insert(n);
        }

        vector<int> ans;
        for (int n : nums2) {
            if (uset1.count(n)) {
                uset2.insert(n);
            }
        }

        for( auto & p : uset2){
            ans.push_back(p);
        }
        
        return ans;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: list[int], nums2: list[int]) -> list[int]:
        # Convert the first list into a set to remove duplicates and enable efficient lookups
        set1 = set(nums1)
        
        # Convert the second list into a set to remove duplicates as well
        set2 = set(nums2)
        
        # Find the intersection of both sets using the bitwise AND operator and convert back to a list
        return list(set1 & set2)